Makefile.in

```diff
@@ -352,28 +352,30 @@ dist: 3-stage-bootstrap
 	cp $(srcdir)/build/packages/unix/README release/opendylan-$(version)/
 	cd release && tar cjf opendylan-$(version)-$(TARGET_PLATFORM).tar.bz2 opendylan-$(version)
 
-CHECK_APPS = \
-	libraries-test-suite-app \
-	dfmc-reader-test-suite-app \
-	strings-test-suite-app \
-	c-ffi-test-suite-app \
-	duim-test-suite-app \
-	deuce-test-suite-app \
-	jam-test-suite-app \
+
+TEST_LIBS = \
+	libraries-test-suite \
+	dfmc-reader-test-suite \
+	strings-test-suite \
+	duim-test-suite \
+	deuce-test-suite \
+	jam-test-suite \
 	channels-tests \
 	variable-search-tests \
-	command-line-parser-test-suite-app \
-	logging-test-suite-app \
+	command-line-parser-test-suite \
+	logging-test-suite \
 	regular-expressions-test-suite \
 	hash-algorithms-test
 
+TEST_LOADS = $(shell for lib in $(TEST_LIBS); do echo "--load lib$${lib}.so "; done)
```

Comment:
The following change to configure.ac will add a substitution variable for shared library extension, which you can then reference as `@shrext@` in Makefile.in:

```diff
diff --git a/configure.ac b/configure.ac
index ff5baa80..d0f38d0e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -13,6 +13,7 @@ MACOS_SDK=
 
 AC_CANONICAL_TARGET
 
+shrext=.so
 MULTIARCH=$host
 case $host in
   i*86-*-linux*)
@@ -31,11 +32,13 @@ case $host in
       OPEN_DYLAN_TARGET_PLATFORM=x86-darwin;
       MACOS_SDK=`xcrun --show-sdk-path`
       SUPPORT_LLVM=yes
+      shrext=.dylib
     ;;
   x86_64-*-darwin*)
       OPEN_DYLAN_TARGET_PLATFORM="x86_64-darwin";
       MACOS_SDK=`xcrun --show-sdk-path`
       SUPPORT_LLVM=yes
+      shrext=.dylib
     ;;
   amd64-*-freebsd* | x86_64-*-freebsd*)
       OPEN_DYLAN_TARGET_PLATFORM=x86_64-freebsd;
@@ -68,6 +71,7 @@ AC_SUBST(OPEN_DYLAN_TARGET_PLATFORM)
 AC_SUBST(ADDITIONAL_RUNTIME_TARGET_PLATFORMS)
 
 AC_SUBST(MACOS_SDK)
+AC_SUBST(shrext)
 
 AM_INIT_AUTOMAKE
 
```

Reply:
done